client/state/jetpack-connect/actions/authorize.js

```diff
@@ -67,23 +67,7 @@ export function authorize( queryObject ) {
 				} );
 
 				// Update the user now that we are fully connected.
-				const user = userFactory();
-				user.fetching = false;
-				user.fetch();
-
-				// @TODO: When user fetching is reduxified, let's get rid of this hack.
-				// Currently, we need it to make sure user has been refetched before we continue.
-				// Otherwise the user might see a confusing message that they have no sites.
-				// See p8oabR-j3-p2/#comment-2399 for more information.
-				return new Promise( ( resolve ) => {
-					const userFetched = setInterval( () => {
-						const loadedUser = user.get();
-						if ( loadedUser ) {
-							clearInterval( userFetched );
-							resolve( loadedUser );
-						}
-					}, 100 );
-				} );
+				return dispatch( fetchCurrentUser() );
```

Comment:
The old a new code behave a bit differently:
- old code would send the request even if one was already in flight. By setting `user.fetching` to `false` explicitly. The new code won't issue a second request.
- if the fetch fails, the promise in the old code never resolves because `user.get()` is never truthy. The new code will proceed even if the fetch fails, ignoring the error.

Are these differences important? The first one probably isn't, but the fact that `fetchCurrentUser` never returns an error is a shortcoming.

Reply:
Regarding the first one - I wasn't able to locate an instance where we were running a `/me` request and had to abort it to re-request it. In any case, having competing identical requests could cause weird race conditions, so I'd very much prefer to avoid that intentionally. It's likely an artifact from a previous version of how Jetpack Connect used to work in the pre-Redux age.

Regarding the second one - since `dispatch( fetchCurrentUser() )` will return the thunk, if that fails, won't it be caught by the `.catch()` below? In which scenario, we'd expectedly end up with an authorization error.

Reply:
`fetchCurrentUser()` will return the `fetchingUser` promise which never fails: it's a result of `wpcom.me().get().then().catch().finally()`. The potential error has already been caught. The promise is always `Promise.resolve( undefined )`.

It's not a blocker for this PR, but something we should consider changing in the `fetchCurrentUser` implementation. Currently it swallows and ignores the errors, which may not be always desirable.